tajo-storage/tajo-storage-hdfs/src/main/java/org/apache/tajo/storage/FileTablespace.java

```diff
@@ -550,6 +551,181 @@ public long getMinSplitSize() {
+    for (Path p : inputs) {
+      ArrayList<FileStatus> files = Lists.newArrayList();
+      if (fs.isFile(p)) {
+        files.addAll(Lists.newArrayList(fs.getFileStatus(p)));
+      } else {
+        files.addAll(listStatus(p));
+      }
+
+      int previousSplitSize = splits.size();
+      for (FileStatus file : files) {
+        Path path = file.getPath();
+        long length = file.getLen();
+        if (length > 0) {
+          // Get locations of blocks of file
+          BlockLocation[] blkLocations = fs.getFileBlockLocations(file, 0, length);
+          boolean splittable = isSplittable(meta, schema, path, file);
+          if (blocksMetadataEnabled && fs instanceof DistributedFileSystem) {
+
+            if (splittable) {
+              for (BlockLocation blockLocation : blkLocations) {
+                volumeSplits.add(makePartitionSplit(tableName, path, blockLocation, partitionKeys[i]));
+              }
+              blockLocations.addAll(Arrays.asList(blkLocations));
+
+            } else { // Non splittable
+              long blockSize = blkLocations[0].getLength();
+              if (blockSize >= length) {
+                blockLocations.addAll(Arrays.asList(blkLocations));
+                for (BlockLocation blockLocation : blkLocations) {
+                  volumeSplits.add(makePartitionSplit(tableName, path, blockLocation, partitionKeys[i]));
+                }
+              } else {
+                splits.add(makeNonPartitionSplit(tableName, path, 0, length, blkLocations, partitionKeys[i]));
+              }
+            }
+
+          } else {
+            if (splittable) {
+
+              long minSize = Math.max(getMinSplitSize(), 1);
+
+              long blockSize = file.getBlockSize(); // s3n rest api contained block size but blockLocations is one
+              long splitSize = Math.max(minSize, blockSize);
+              long bytesRemaining = length;
+
+              // for s3
+              while (((double) bytesRemaining) / splitSize > SPLIT_SLOP) {
+                int blkIndex = getBlockIndex(blkLocations, length - bytesRemaining);
+
+                splits.add(makePartitionSplit(tableName, path, length - bytesRemaining, splitSize,
+                  blkLocations[blkIndex].getHosts(), partitionKeys[i]));
+
+                bytesRemaining -= splitSize;
+              }
+              if (bytesRemaining > 0) {
+                int blkIndex = getBlockIndex(blkLocations, length - bytesRemaining);
+                splits.add(makePartitionSplit(tableName, path, length - bytesRemaining, bytesRemaining,
+                  blkLocations[blkIndex].getHosts(), partitionKeys[i]));
+              }
+            } else { // Non splittable
+              splits.add(makeNonPartitionSplit(tableName, path, 0, length, blkLocations, partitionKeys[i]));
+            }
+          }
+        }
+      }
+      if(LOG.isDebugEnabled()){
+        LOG.debug("# of splits per partition: " + (splits.size() - previousSplitSize));
```

Comment:
The log message is somewhat different from the actual meaning.

``PreviousSplitSize`` is the number of files in partition directories, and split is the number of total fragments. For example, there are two files in all partitions, and each file makes 5 fragments. Then, the total number of fragments is 10. In that case, 10 - 2 is jut 8. 8 is not the number of splits per partition.  The actual answer should be 5. In addition, we can say X per Y when X is the same across all Ys. It is hard to get answer because each partition file can be vary in terms of size. So, average is more proper here, I think.